### ISSUE TITLE
Project dead?

### ISSUE THREAD

**Issue by youurayy** [axios/axios#1965]
Is the project dead? Are new maintainers sought?

- 97 open pull requests
- 411 open issues

**Comment by likern** [comment]
And no commits in master since September. No plans to support HTTP2 yet.

**Comment by emilyemorehouse** [comment]
It's not dead, I just haven't been able to personally do as much on the project lately. We had a big issue with fixing configurations, which introduced breaking changes, that have halted things until that gets fixed. 

So yes, if there are people willing to step up and help as maintainers, I welcome them!

**Comment by youurayy** [comment]
@emilyemorehouse maybe it would be a good idea to advertise that in the README.md, I'm sure there are folks who want to help.

**Comment by rogeriopvl** [comment]
@emilyemorehouse I would be glad to help out as a maintainer

**Comment by 43081j** [comment]
Haven't used the project before admittedly but usually found digging around github, so happy to have a read through some PRs in my spare time.

What would be super nice is a new version which uses ES modules 🙊

**Comment by iidrees** [comment]
Hi @emilyemorehouse, I would love to help out.

**Comment by ironicnet** [comment]
I think that doing triage or labeling the issues would help the new volunteers to know where to start...

**Comment by hugomd** [comment]
I’d be keen on helping out, whether it’s triaging, actively contributing, or anything else 🤙

**Comment by lnghrdntcr** [comment]
@emilyemorehouse I'd love to help! Probably would be nice to have a some sort of tags (like, 'for beginners' or similar) on the issues in order to encourage contributors!

**Comment by rajasekarm** [comment]
I’m using axios for more than a year. I can help in fixing bugs.

**Comment by gkarwchan** [comment]
I would love to contribute. I used axios in 4 projects so far, and it is my first choice always.

**Comment by GlebkaF** [comment]
I could help, just tell me what to start with :)

**Comment by eranshapira** [comment]
count me in.

**Comment by simsim0709** [comment]
As the heavy user of axios, it would be great if I can help out. :)

**Comment by 3sanket3** [comment]
I don't have much contribution history but I use axios in projects. I will happy to help.

**Comment by kofoworola** [comment]
As an axios junkie, i'd love to help however i can

**Comment by luciano7** [comment]
@mzabriskie @emilyemorehouse please let me know how we can help!

**Comment by cdcasey** [comment]
I would love to help.

**Comment by wlepinski** [comment]
I would love to help as well.

**Comment by nickstew** [comment]
I'd be happy to assist in maintaining this project. I've been using it for years and would be happy to lend a hand, @emilyemorehouse. :heart:

**Comment by bmacher** [comment]
I would help aswell. :)

**Comment by nolleto** [comment]
I would like to help too 😄

**Comment by atishaybaid** [comment]
I would like to contribute and maintain ,let me know where to start

**Comment by varijkapil13** [comment]
I am using axios since along time and in multiple projects,, I would love to help any way possible

**Comment by bipin000** [comment]
will be my pleasure to contribute to axios

**Comment by mowatermelon** [comment]
I would love to help as well.

**Comment by Eqls** [comment]
Would love to start contributing, been using it for a year so far for all of my projects and I want to help to keep it healthy.

Some type of guidance on where to start would be appreciated though!

**Comment by vonvick** [comment]
Would love to help as I have used axios extensively.

**Comment by emilyemorehouse** [comment]
😭 y'all are AWESOME.

To anyone who wants to help, here are a few ideas I have:

- **Triage issues:** I recently added issue templates to help auto-tag issues (and filter out actual bugs vs usage issues). There's _a lot_ of noise for this project and I spend the majority of my time trying to filter through issues and wind up closing most of them with a simple "This doesn't seem like an Axios bug (many I can't even duplicate), I think X may be your issue, feel free to post on Gitter or Stack Overflow for help debugging your code". If you find a real bug that doesn't have example code, providing example code is a HUGE help. Bonus points if it's as simple as copy/pasting into [Runkit](https://npm.runkit.com/axios) with calls to an example API like [JSON Placeholder](https://jsonplaceholder.typicode.com). 
- **PR Review:**  Not quite as noisy as issues, but this can still be a lot to go through. I really appreciate people who tag me in PRs that have high priority/fix known issues. Feel free to ping me if I don't respond after a few days. Currently, the focus is definitely getting things stable before focusing on new features or [1.0.0](https://github.com/axios/axios/projects/2).
- **CI:** Our CI is finicky - we often hit weird edge cases or issues that cause CI to break and that slows up the whole procress. If we have a broken master branch, I can't release, plain and simple. So if you ever see that master is failing (or PRs are failing for issues not caused by the PR), any help there is massively appreciated.

I'm happy to give anyone access as needed. The only thing I'd like to hold onto is acting as the release manager to ensure consistency.

I plan on adding this info to the contributing doc along with my response templates for others to use and guidelines for how issues should be labeled, etc.

**Comment by luciano7** [comment]
I posted a request for help on reddit (https://www.reddit.com/r/javascript/comments/an94xq/axios_needs_maintainers/) - as you can see, your work has a lot of fans. I look forward to start contributing and once again- thank you.

**Comment by octalpixel** [comment]
@emilyemorehouse would love to help out.

**Comment by DarkSeidNG** [comment]
@emilyemorehouse I would love to help out

**Comment by luckyincode** [comment]
@emilyemorehouse I too would love to help!

**Comment by chinesedfan** [comment]
It seems like a contradiction.

- Without contributions, @emilyemorehouse can't distinguish helpful people and grant permissions.
- Without permissions, people usually won't start to contribute.

I suggest everyone try to subscribe as [code helpers](https://www.codetriage.com/axios/axios) first. If we can persist in triaging issues or sending pull requests, @emilyemorehouse will be more glad to see. And one day, you will be a maintainer naturally.

**Comment by Sequoia** [comment]
I use axios and am not currently concerned about lack of ongoing development (tools typically don't blow up the moment a maintainer steps away), but if anyone is really raring to go & wants to start merging PRs, note that this is an open source project and that there's no cost (besides time) to forking & start reviewing and merging PRs to you fork. This can be a natural progression in the life of an open source tool.

Not saying "go fork it," but "if you're feeling frustrated or feeling like the project is 'dead', you don't need permission to bring it back to life. The power is yours! 🌍 "

**Comment by askiiRobotics** [comment]
Probably it would be nice to have a discord for futher cooperation in the topic.

**Comment by AllainPL** [comment]
@emilyemorehouse @mzabriskie @nickuraltsev @rubennorte - What are your plans with axios? Last published beta version was 8 months ago, stable a year ago, you get 4,5 milion downloads a week. It's just a milion less than e.g. react. Taking into consideration this popularity it would be really cool to know what is the status and for you maybe to take some help from the community :)

@Sequoia you are quite right however it's kind of difficult at this point when there are so many open issues and PR's that it's even difficult to decide which are still valid. @emilyemorehouse wrote 2 months ago about how we can help - the first thing is to go over all the issues and do some cleaning :)

**Comment by stanmots** [comment]
It's sad to see such a nice project slowly dying. It was literally the best http client I used in nodejs.

**Comment by pchaganti** [comment]
👍

**Comment by igorshubovych** [comment]
@emilyemorehouse 
I was once core team in tldr-pages. At that time project author had some family issues and my 2nd kid was born. I understood that we will not do it ourselves and we added 3 new members to core team to help us review PRs.

You are working in quite an important project for the internet. It should not be abandoned. There are 8 proposed PRs in the last month, none of them was merged. More importantly there is a known security vulnerability published (https://snyk.io/vuln/SNYK-JS-AXIOS-174505), there is PR #1485 also not merged.

With that amount of PRs it would be better if you guys start accepting help. There are many people who would be proud to be the maintainer of such project. Some of them already proposed help in this thread.

Good luck!

**Comment by Ryuno-Ki** [comment]
> * **Triage issues:** I recently added issue templates to help auto-tag issues (and filter out actual bugs vs usage issues). There's _a lot_ of noise for this project and I spend the majority of my time trying to filter through issues and wind up closing most of them with a simple "This doesn't seem like an Axios bug (many I can't even duplicate),

Maybe you could adopt [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736)?
(Thanks, @zachleat for promoting this in @11ty/eleventy!)

**Comment by igorshubovych** [comment]
@emilyemorehouse 
I see you merged 6 PRs (3 of them are for README.md). This is great start.

But honestly, I still believe you need to add collaborators to the project, because you have a lot of work.

1. As you merged security fix, you need to do an immediate release (something like v0.18.1).
1. You have 100+ PRs. Team of collaborators needs to review them and merge / decline.
1. Axios has 500+ issues. Many of them are duplicates or not relevant. Collaborators team need to regularly review them and keep the amount of open issues low, or at least managable.
1. And more importantly, your team needs to have regular routine on reviewing PRs and issues, so the situation will not repeat.

Please start accepting help, otherwise the project will become dead. Best wishes!

**Comment by michaelvdnet** [comment]
I think we can divide workload by doing code reviews on PRs. You dont need to be a collaborator for that and reviewing is a time intensive task.

We still need people to merge PRs though.

**Comment by NicoZelaya** [comment]
I'm willing to chip in too but I'm not certain on what do we need to unblock the release it's cooking (which includes the DoS vulnerability fix, an important one to release). 

Please, if there's any way I can help here, reach out!

**Comment by jasonsaayman** [comment]
I am also willing to get involved, we use axios in a number of extremely critical projects. It would be terrible if axios were to not be supported in the future.

**Comment by igorshubovych** [comment]
@emilyemorehouse
@mzabriskie 
@nickuraltsev
@rubennorte
could you please discuss adding more collaborators to the project?

**Comment by Sequoia** [comment]
This issue having been open for 4 months now, I'm inclined to agree with the complainers here at this point. Good stewardship of this project would look like either:

a) picking from among the many willing people to add more contributors, or
b) marking the project as archived to make maintenance intentions clear & allowing people to move on to next steps or
c) maintainers step back in to continue active maintenance

Maintainers: it would be a great service to the community of users if you would please pick one of the above option so those interested in maintaining this project can move forward. Archiving is perfectly acceptable :+1:

**Comment by mapleeit** [comment]
I saw that @emilyemorehouse merged some PR yesterday. So it seems there are still somebody working on this repo. That's really great.

So did you guys finish with the "big configuration issue"? When will the next version be released?

Thanks for you effort.

**Comment by leslie-alldridge** [comment]
I love axios, keen to help however I can

**Comment by robmcguinness** [comment]
🙏 https://github.com/axios/axios/releases/tag/v0.19.0

**Comment by igorshubovych** [comment]
Hats off to @emilyemorehouse for making a release.
I hope there will be a possibility to do regular releases soon.

**Comment by jphilipstevens** [comment]
@emilyemorehouse

I would also like to.help out. At my work we use Axios quite a bit and I'm more than willing to spend time on this project

**Comment by ouraios** [comment]
ping @rubennorte @mzabriskie  @emilyemorehouse @nickuraltsev   

Lots of people using your amazing package are waiting from you to take some actions to take some more hands to help you !   

And if you dont have time to do those actions then maybe you could at least label this package as dead ?  

Maybe setting up a patreon could help you to be motivated to start working on it again ?

**Comment by akx** [comment]
I wouldn't mind helping either, e.g. in a triage role (https://github.blog/changelog/2019-05-23-triage-and-maintain-roles-beta/) :)

**Comment by thexpand** [comment]
I think that was a begged release, because I don't see any activity after the latest 0.19.0 release.
Such a shame for such an awesome HTTP client.

**Comment by snowdream** [comment]
Is there any plan to add HTTP/2 support for node.js?   https://github.com/axios/axios/issues/1175

**Comment by igorgue** [comment]
Looks like a pretty alive project to me...

**Comment by akx** [comment]
> Looks like a pretty alive project to me...

There was an entirely quiet period from May to September, if you look closely.

Either way I'm happy momentum is picking up again. :)

**Comment by unguul** [comment]
With a reported usage of 1.8m you'd expect this package to receive some kind of financial backing. I see no patreon, opencollective, tidelift or [insert income sources here] mentioned anywhere. However, I do see https://opencollective.com/axios but the people listed are not part of the axios project (https://github.com/orgs/axios/people) so maybe someone from the axios org should take a look. I created #2493 for follow up.

**Comment by viT-1** [comment]
What about esm version?
https://github.com/axios/axios/issues/1879

**Comment by BrunnerLivio** [comment]
The Googleapis organization has started a new project called [gaxios](https://github.com/googleapis/gaxios)

The reason why from issue [googleapis/gaxios#81](https://github.com/googleapis/gaxios/issues/81)

> We were originally using axios, but created this for a few reasons:
> - Axios has been relatively unmaintained recently
> - It didn't support proxy support out of the box
> - It didn't support retry out of the box
> - It duplicated a lot of client side logic we just didn't need here

> Axios is a great library, but we wanted something built on the much lighter node-fetch. Hope this helps!

I understand the motivation behind Googleapis team, though I feel like it is not a good idea to create a new library which are so similar {insert xkcd comic}. May we can find common ground here and just let it be one “Axios”? cc @JustinBeckwith

**Comment by chinesedfan** [comment]
Share a personal story and some news for users in this thread,

- I noticed the situation of axios like you guys in Feb 2019 and triaged about 30 issues+pulls.
- Got no responses from the maintainers and felt a little disappointed. But still kept an eye on it.
- Luckily, I was invited as a collaborator in Dec 2019. That started my issue killer time. Total issues have decreased from ~600 to less than 300 now.

I created a plan for the next release. See https://github.com/axios/axios/projects/4. I know there are still some problems in axios, but I hope we can fix those hottest first.

I don't need your words to pay too much time on it, but would someone mind helping review those features when you are free, to make the next release better?

**Comment by dpikt** [comment]
@chinesedfan that's great news, glad to hear things are moving! Question since you're now a collaborator- have plans/timeline coalesced around a 1.0 release? Or will upcoming releases all be 0.x for the time being?

Apologies in advance if this has been answered in a different thread.

**Comment by jasonsaayman** [comment]
Hi,

I think that the v1.0 release as it is currently pinned is so out of date with what it was trying to achieve that I almost want to say its technical debt. The goals of v1.0 should be clearly laid out and evaluated as to what is most important to the community.

I think we should be looking towards stability more than anything else with proper regression testing. Axios needs to also have a stable release strategy that should be locked down in a v1.0 with a more structured pre-release with testing strategy.

I will endeavour to try get a new v1.0 project up with issues etc. very soon.

**Comment by Ryuno-Ki** [comment]
Given that the maintenance question seem to be solved by now … can we close this issue/lock it?

**Comment by jasonsaayman** [comment]
Closing for now, please watch projects to see new versions and proposed road map for axios.
